src/runtime/cli/config/configuration-qemu.toml.in

```diff
@@ -527,3 +534,27 @@ experimental=@DEFAULTEXPFEATURES@
 # If enabled, user can run pprof tools with shim v2 process through kata-monitor.
 # (default: false)
 # enable_pprof = true
+
+[image]
+# Container image service.
+#
+# Offload the CRI image management service to the kata agent.
+service_offload = true
+
+# Container image decryption keys provisioning.
+# Applies only if service_offload is true.
+# Keys can be provisioned locally (e.g. through a special command or
+# a local file) or remotely (usually after the guest is remotely attested).
+# The provision setting is a complete URL that lets the Kata agent decide
+# which method to use in order to fetch the keys.
```

Comment:
I don't think that the method to fetch the keys should be stored on the host. That is probably something that belongs to the (encrypted / attested) `initrd` image.

What rightfully belong to the host is describing the key provisioning and attestation method that is available on this hardware. We can either do it by describing the platform, i.e. `secure_platform = TDX`, or with a more descriptive approach, e.g. `attestation = remote`.

Reply:
cc @sameo @bpradipt  ^^

Reply:
> I don't think that the method to fetch the keys should be stored on the host. That is probably something that belongs to the (encrypted / attested) `initrd` image.
> 

If `initrd` is encrypted, then key provisioning and thus attestation will belong to the firmware layer, which then needs to be attested but not encrypted. Your key provisioning method will be stored there, in an unencrypted firmware image stored on the host. There's not much difference between this and storing it in a configuration file. Hiding it somewhere in a binary initrd or a firmware file won't make things more secure imho.

> What rightfully belong to the host is describing the key provisioning and attestation method that is available on this hardware. We can either do it by describing the platform, i.e. `secure_platform = TDX`, or with a more descriptive approach, e.g. `attestation = remote`.

There is not always a direct mapping between the attestation method and the platform implementation.

Reply:
@c3d the option in the config will be used by the kata agent to execute the platform specific method to get the decryption key for container (or other secrets).  Do you see any issues ?

Reply:
@c3d  does it make sense?  what do you think ?

Reply:
same here